app/src/main/java/com/mlinde/marvelcomicsapp/data/ComicsRepository.kt

```diff
@@ -6,31 +6,7 @@ import kotlin.Exception
 
 class ComicsRepository(private val apiService: ApiService) {
 
-    suspend fun getComics(): ApiRensponse {
-        return try {
-            //val response = ComicsService.apiService.comics()
-            val response = apiService.comics()
-            if (response.isSuccessful) {
-                ApiRensponse.Success(response.body())
-            } else {
-                ApiRensponse.Error("Error")
-            }
-        } catch (exception: Exception) {
-            ApiRensponse.Error("Error")
-        }
-    }
+    suspend fun getComics() = apiService.comics()
```

Comment:
You should check if response is successful. Something like that:
```
if (!response.isSuccessful) {
   throw HttpException(response)
 }
 return response 
 ```
 Then in your view model `onFailure ` will catch it.